File: Java/DataStructures/MyStack.java
package Java.DataStructures;

public class MyStack{

    private int stack[];
    private int capacity;
    private int top;

    public MyStack(int size){
        stack = new int[size];
        capacity = size;
        top = -1;
    }

    public void push(int item){

        if(isFull()){
            throw new RuntimeException("Stack is full");
        }

        top++;
        stack[top] = item;
    }

    public int pop(){

        if(isEmpty()){
            throw new RuntimeException("Stack is empty");
        }

        int topElement = stack[top];
        top--;

        return topElement;
    }

    public Boolean isEmpty(){
        return top == -1;
    }

    public Boolean isFull(){
        return top == capacity - 1;
    }

    public int peek(){

        if(isEmpty()){
            throw new RuntimeException("Stack is empty");
        }

        return stack[top];
    }

    public void printStack(){
        String stackAsString = "[";

        if(!isEmpty()){
            for(int i = 0; i <= top; i++){
                if(i == 0){
                    stackAsString = stackAsString  + Integer.toString(stack[i]);
                }else{
                    stackAsString = stackAsString + " | " + Integer.toString(stack[i]);
                }
            }
        }

        stackAsString = stackAsString + "] <- Top";
        System.out.println(stackAsString);
    }
}

File: Python/Data Structures/MyStack.py
class MyStack:

    def __init__(self, size):
        self.capacity = size
        self.top = -1
        self.stack = [None] * size

    def push(self, item):

        # Checking if the stack is fulll
        if self.isFull():
            raise Exception(f'The element {item} could not be pushed on to the stack because the stack is full.')

        self.top = self.top + 1
        self.stack[self.top] = item
            
    def pop(self):

        if self.isEmpty():
            raise Exception('Nothing to pop. Stack is empty.')

        topElement = self.stack[self.top]
        self.top = self.top - 1

        return topElement

    def isEmpty(self):
        return self.top == -1

    def isFull(self):
        return self.top == self.capacity - 1

    def peek(self):
        if self.isEmpty():
            raise Exception('Nothing to peek. Stack is empty.')

        return self.stack[self.top]

    def __str__(self):

        if self.isEmpty():
            return '[]'

        return f'{[self.stack[x] for x in range(self.top + 1)]} <- Top'